server/controllers/appointments/appointmentsSearch.js

```diff
@@ -0,0 +1,39 @@
+const { getAppointmentsByPatientName } = require('../../database/queries');
+const { appointmentsSearchSchema } = require('../../utils/validation');
+const { boomify } = require('../../utils');
+
+const appointmentsSearch = async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      phone,
+    } = await appointmentsSearchSchema.validate(req.query, {
+      abortEarly: false,
+    });
+
+    if ((firstName && lastName) || phone) {
+      const { rows: data } = await getAppointmentsByPatientName({
+        firstName,
+        lastName,
+        phone,
+      });
+      res.json({
```

Comment:
Alaa,when I am using your routes like this :-
http://localhost:5000/api/v1/appointments/search?firstName=Alexie&lastName=Jenkins
I got 3 items but I expected one .

```json
{
StatusCode: 200,
data: [
{
id: 1,
appointment_date: "2021-12-02T00:00:00.000Z",
appointment_time: "22:30:00",
firstname: "Alexie",
lastname: "Jenkins"
},
{
id: 4,
appointment_date: "2021-12-02T00:00:00.000Z",
appointment_time: "22:30:00",
firstname: "Alexie",
lastname: "Jenkins"
},
{
id: 8,
appointment_date: "2020-12-02T00:00:00.000Z",
appointment_time: "20:30:00",
firstname: "Alexie",
lastname: "Jenkins"
}
]
}
```